### PR TITLE
Enhancement: Add sysdig agent tags

### PR DESCRIPTION
**Pull request by dishankkalra23**
### Description

Add sysdig tags in the TIM observability module such that it could be consumed by TIM OCP-all-inclusive and GE-Observability module.

GE Issue: https://github.ibm.com/GoldenEye/issues/issues/6179
PR: https://github.ibm.com/GoldenEye/ocp-all-inclusive-module/pull/427#issuecomment-65598738

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: examples/basic/variables.tf [hunk 1]
```diff
@@ -33,3 +33,9 @@ variable "logdna_agent_tags" {
   description = "Array of string of tags for logdna agent"
   default     = []
 }
+
+variable "sysdig_agent_tags" {
```
Comment (ocofaigh) [inline examples/basic/variables.tf:37]: do not add this to the example - we will use the `resource_tags` variable for tagging all resources provisioned by an example

Comment (ocofaigh) [reply examples/basic/variables.tf:37]: Can you also remove `logdna_agent_tags` from the example - it should not be there

File: examples/basic/main.tf [hunk 2]
```diff
@@ -114,4 +114,5 @@ module "observability_agents" {
   logdna_add_cluster_name   = true
   # example of how to include / exclude metrics - more info https://cloud.ibm.com/docs/monitoring?topic=monitoring-change_kube_agent#change_kube_agent_log_metrics
   sysdig_metrics_filter = [{ type = "exclude", name = "metricA.*" }, { type = "include", name = "metricB.*" }]
+  sysdig_agent_tags     = var.sysdig_agent_tags
```
Comment (ocofaigh) [inline examples/basic/main.tf:117]: update this to `sysdig_agent_tags = var. resource_tags`

can you also update this too:
`logdna_agent_tags = var.resource_tags`

File: variables.tf [hunk 3]
```diff
@@ -105,4 +105,17 @@ variable "sysdig_metrics_filter" {
   }
 }
 
+variable "sysdig_agent_tags" {
+  type        = list(string)
+  description = "array of tags to group the host logs pushed by the sysdig agent"
```
Comment (ocofaigh) [inline variables.tf:110]: sysdig deals with metrics, not logs

File: variables.tf [hunk 4]
```diff
@@ -105,4 +105,17 @@ variable "sysdig_metrics_filter" {
...
+  default     = []
+  nullable    = false
+}
+
+variable "sysdig_add_cluster_name" {
+  type        = bool
+  description = "If true, configure the sysdig agent to attach a tag containing the cluster name to all log messages."
```
Comment (ocofaigh) [inline variables.tf:117]: sysdig deals with metrics, not logs

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

File: variables.tf (hunk 3 above)
Comment (dishankkalra23) [reply variables.tf:110]: I guess it would be just replacing logs with metrics?

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**